[ci]: Update integration test script to be standalone

hickeyma opened the pull request:
~The integration script was exiting after OpenAI server started instead of continuing on and calling the test cases. This fixes the issue by breaking loop testing for server starting (instead of exiting the script) and therefore continues on.~ Fixed in #1090 

Update to the integration test script as follows:

- Make integration script standalone (minus buildkite dependencies) so that it can be called independent of buildkite
- Add variable names for file names for easier reference and cleanup
- Move the script into `scripts` folder in repo root for easier access

~Fixes #1086~ Fixed in #1090

**Note to reviewers**:

PR changed from bug fix to making the script standalone as per discussion in https://github.com/LMCache/LMCache/pull/1087#issuecomment-3097404967.

hickeyma committed:
Fix test cases not running

The integration scripts script was exiting after OpenAI server started
instead of continuing on and calling the test cases. This fixes the issue.

This bug was introduced in #1074.
Fixes #1086

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

Shaoting-Feng commented:
Sorry, Martin—I’ve only just seen this PR. I submitted a similar PR #1090 fixing issue #1086. Could you please rebase this PR if necessary? Apologies for the extra effort.

hickeyma commented:
> Sorry, Martin—I’ve only just seen this PR. I submitted a similar PR https://github.com/LMCache/LMCache/pull/1090 fixing issue https://github.com/LMCache/LMCache/issues/1086. Could you please rebase this PR if necessary? Apologies for the extra effort.

That's ok @Shaoting-Feng. Thanks for pushing the fix.

Re changes to the integration tests, I see that you have moved some cleanup to https://github.com/LMCache/LMCache/blob/dev/.buildkite/clean.yaml and also moved the integration tests script to within `.buildkite` folder in https://github.com/LMCache/LMCache/tree/dev/.buildkite. Is this because these scripts are run in the buildkite pipeline? I am asking because would it be better to keep the integration tests standalone in the 1 script in `scripts` folder in project root?  This way not only can the buildkite pipeline uses it but others like users who want to run the tests can do also?

Shaoting-Feng commented:
> I am asking because would it be better to keep the integration tests standalone in the 1 script in `scripts` folder in project root?

Sounds good to me. We should put them in a separate folder. Also, please resolve the conflicts, as I think there is still one left.

hickeyma commented:
> Sounds good to me. We should put them in a separate folder. Also, please resolve the conflicts, as I think there is still one left.

Thanks @Shaoting-Feng, I will do that. What conflicts are you referring to?

Shaoting-Feng commented:
Thanks! I mean the conflict in `vllm-integration-tests.sh`.
<img width="917" height="286" alt="Screenshot 2025-07-22 at 9 35 30 AM" src="https://github.com/user-attachments/assets/7c1e0656-b0f0-4ce0-9b9d-2cd8dcb0f9fb" />

hickeyma committed:
Make the integration tests standalone

Make the integration tests runnable outside of buildkite.
This enables user to run the tests on their own systems.

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma committed:
Merge branch 'dev' into ci/add-fix-integ-test

hickeyma committed:
Merge branch 'dev' into ci/add-fix-integ-test

hickeyma commented:
Lots of great updates from @Shaoting-Feng, @sammshen and others that have improved CI/CD and automated testing in test environments. Closing for now as this has meant lots of test environment changes becoming part of thew script. It might be worth stepping back for now and deciding if want to make it standalone again in the future.
